.github/workflows/run-forge.yaml

```diff
@@ -86,21 +86,19 @@ jobs:
           * [Test runner output](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})
           EOF
 
+          # add github step summary as described here https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary
+          echo "$FORGE_COMMENT" >> $GITHUB_STEP_SUMMARY
+
           # export env vars for next step
-          echo "FORGE_EXIT_CODE=$FORGE_EXIT_CODE" >> $GITHUB_ENV
           echo "FORGE_COMMENT=$FORGE_COMMENT" >> $GITHUB_ENV
 
+          if [ "$FORGE_BLOCKING" = "true" ]; then
+            exit $FORGE_EXIT_CODE
+          fi
+
       - name: Post forge result comment
-        if: env.FORGE_ENABLED == 'true' && github.event.number != null
+        if: env.FORGE_ENABLED == 'true' && github.event.number != null && !cancelled()
```

Comment:
Why do we need this additional check now?

Reply:
if we don't specify `!cancelled()` it implicitly adds a `succeeded()` which means this step only runs when the previous step succeed. We want however to run this "report" step regardless of whether the previous step (forge) failed or succeeded.